app/models/miq_provision.rb

```diff
@@ -59,11 +59,16 @@ def placement_auto
   end
 
   def after_request_task_create
-    vm_name                      = get_next_vm_name
-    options[:vm_target_name]     = vm_name
-    options[:vm_target_hostname] = get_hostname(vm_name)
-    self.description             = self.class.get_description(self, vm_name)
-    save
+    update_vm_name(get_next_vm_name, :update_request => false)
+  end
+
+  def update_vm_name(new_name, update_request: true)
+    new_name = get_vm_full_name(new_name, true)
```

Comment:
Should we call the class method directly?  (even though it seems strange that it is a class method)

Reply:
Sure, I can remove the instance method.  It is a class method because the logic used from both `MiqRequest` and `MiqRequestTask`.